internal/resources/konnectivity/agent.go

```diff
@@ -175,6 +182,12 @@ func (r *Agent) mutate(ctx context.Context, tenantControlPlane *kamajiv1alpha1.T
 		args["--health-server-port"] = "8134"
 		args["--service-account-token-path"] = "/var/run/secrets/tokens/konnectivity-agent-token"
 
+		extraArgs := utilities.ArgsFromSliceToMap(tenantControlPlane.Spec.Addons.Konnectivity.KonnectivityAgentSpec.ExtraArgs)
+
+		for k, v := range extraArgs {
+			args[k] = v
+		}
```

Comment:
This would allow users to shoot in their feet, such as overriding the following flags that must be controlled:
- `service-account-token-path`
- `health-server-port` 
- `admin-server-port`
- `ca-cert`

My I ask you which values are you trying to override? Just to understand the rationale.

Reply:
As discussed, LGTM: please, just document better in the `extraArgs` specification that __hic sunt leones__, so in case of wrong values the konnectivity server/agent components could misbehave.